src/plone/volto/slate/html2slate.py

```diff
@@ -0,0 +1,398 @@
+
+        fragments = fragments_fromstring(text)
+        nodes = []
+        for f in fragments:
+            slate_nodes = self.deserialize(f)
+            if slate_nodes:
+                nodes += slate_nodes
+
+        return self.normalize(nodes)
+
+    def deserialize(self, node):
+        """Deserialize a node into a list Slate Nodes"""
+
+        if node is None:
+            return []
+
+        if is_textnode(node):
+            text = collapse_inline_space(node)
+            return [{"text": text}] if text else None
+        elif not is_element(node):
+            return None
+
+        tagname = node.name
+        handler = None
+
+        if "data-slate-data" in node.attrs:
+            handler = self.handle_slate_data_element
+        else:
+            handler = getattr(self, "handle_tag_{}".format(tagname), None)
+            if not handler and tagname in ACCEPTED_TAGS:
+                handler = self.handle_block
+
+        if handler:
+            slate_node = handler(node)
+            if not isinstance(slate_node, list):
+                slate_node = [slate_node]
+            return slate_node
+
+        # fallback, "skips" the node
+        return self.handle_fallback(node)
+
+    def deserialize_children(self, node):
+        """deserialize_children.
+
+        :param node:
+        """
+
+        res = []
+
+        for child in node.children:
+            b = self.deserialize(child)
+            if isinstance(b, list):
+                res += b
+            elif b:
+                res.append(b)
+
+        return res
+
+    def handle_tag_a(self, node):
+        """handle_tag_a.
+
+        :param node:
+        """
+        link = node["href"] if "href" in node.attrs else None
+
+        element = {
+            "type": "link",
+            "children": self.deserialize_children(node),
+        }
+        if link is not None:
+            element["data"] = {"url": link}
+
+        return element
+
+    def handle_tag_br(self, node):
+        """handle_tag_br.
+
+        :param node:
+        """
+        return {"text": "\n"}
+
+    def handle_block(self, node):
+        """handle_block.
+
+        :param node:
+        """
+        return {"type": node.name, "children": self.deserialize_children(node)}
+
+    def handle_tag_b(self, node):
+        # TO DO: implement <b> special cases
+        return self.handle_block(node)
+
+    def handle_slate_data_element(self, node):
+        """handle_slate_data_element.
+
+        :param node:
+        """
+        data = node["data-slate-data"]
+        element = json.loads(data)
+        element["children"] = self.deserialize_children(node)
+        return element
+
+    def handle_fallback(self, node):
+        """Unknown tags (for example span) are handled as pipe-through"""
+        return self.deserialize_children(node)
+
+    def normalize(self, value):
+        """Normalize value to match Slate constraints"""
+
+        assert isinstance(value, list)
```

Comment:
If this isn't the case, the program will end with AssertionError: False which won't be super helpful. If this is something that should never happen, using type annotations and static analysis in your editor might be a better way to go. If we expect that we might actually get a list here from certain user input, then we should give the user a more helpful error about what they did wrong.